docs/extensions/tutorials/_posts/1970-01-05-Installing3rdPartyPackages.md

```diff
@@ -0,0 +1,323 @@
+fs.writeFileSync(infoPlistPath, plist.build(infoPlist));
+
+const exec = require('child_process').execSync;
+
+const dependenciesToLink = ['react-native-camera'];
+
+const command = 'node node_modules/react-native/local-cli/cli.js link';
+
+dependenciesToLink.forEach((dependency) => {
+  exec(`${command} ${dependency}`);
+});
+```
+
+Now let's explain the details regarding native dependency linking in `app/scripts/run.js`
+  - everything regarding `Info.plist` is added to notify the user that the app is using his camera
+  - `dependenciesToLink` is an array that stores all of our native dependencies
+  - `command` represents the react-native-cli, used to link native dependencies
+  - the final three lines of code make sure each dependency in our array is linked
+
+The reason we create an array is because sometimes our extension will have multiple native dependencies, so instead of making a separate linking command for each, we simply run one `forEach` loop and hand it an array of dependencies.
+
+### Using the Package
+
+Our extension will now have access to everything react-native-camera has to offer. For this example we'll edit `app/screens/QRReaderScreen.js` so that it displays an alert when the camera reads a QR code and the alert message will contain the QR code data.
+
+```javascript{2-4,6-10,12-21,25-28}
+#file: app/screens/QRReaderScreen.js
+import React, { Component } from 'react';
+import { Alert } from 'react-native';
+import Camera from 'react-native-camera';
+import _ from 'lodash';
+
+export default class QRReaderScreen extends Component {
+    constructor(props) {
+      super(props);
+      this.onBarCodeRead = this.onBarCodeRead.bind(this);
+    }
+
+  onBarCodeRead(code) {
+      Alert.alert(
+        'QR Code Detected',
+        code.data,
+        [
+          {text: 'OK, read it.', onPress: () => console.log('User saw QR Code contents.')},
+        ],
+        { cancelable: false }
+      )
+  }
+
+  render() {
+    return (
+      <Camera
+        onBarCodeRead={_.debounce(this.onBarCodeRead, 1000, { leading: true, trailing: false })}
+        style={% raw %}{{{% endraw %} flex: 1 }}
+      />
+    );
+  }
+}
+```
+
+After making these changes, we can push the extension to Shoutem and install it in a new app. Continuing from where we stopped in our terminal:
+
+```ShellSession
+$ cd ../..
+$ shoutem push
+Uploading `QR Reader` extension to Shoutem...
+Success!
+$ shoutem install --new QRReader
+Extension installed
+See it in browser: `{{ site.shoutem.builderURL }}/app/{{ site.example.appId }}`
+```
+
+Opening the QRReader app in the Builder will show us an app with no Screens, but since we just installed our QR Reader Extension onto the app, we can just add the QRReader screen to it by clicking on the + button next to Screens, going to the Custom category and selecting the QR Reader Extension. Now to test out how it works we'll click "Preview on Device" and use the Shoutem Preview app to scan the QR code.
```

Comment:
Are you sure you can test it on the Shoutem Preview? If so, it's only because this dependency is already installed on the mobilizer. It shouldn't be possible. You need to explain here that it's not possible because you can't change the native code from Shoutem Preview, nor you can preview it in the builder (because, again, native things needs to be changed, which changes app binary). You should tell that they can test their app using `shoutem run-ios` or `shoutem run-android` command which will bundle extensions to app and will install these native dependencies into new app binary when binary is being created.